gravitee-apim-repository/gravitee-apim-repository-api/src/main/java/io/gravitee/repository/management/api/PortalNavigationItemRepository.java

```diff
@@ -23,11 +23,13 @@ public interface PortalNavigationItemRepository extends CrudRepository<PortalNav
     List<PortalNavigationItem> findAllByOrganizationIdAndEnvironmentId(String organizationId, String environmentId)
         throws TechnicalException;
 
-    List<PortalNavigationItem> findAllByAreaAndOrganizationIdAndEnvironmentId(
-        PortalNavigationItem.Area area,
-        String organizationId,
-        String environmentId
-    ) throws TechnicalException;
+    List<PortalNavigationItem> findAllByParentIdAndEnvironmentId(String parentId, String environmentId) throws TechnicalException;
+
+    List<PortalNavigationItem> findAllByAreaAndEnvironmentIdAndParentIdIsNull(PortalNavigationItem.Area area, String environmentId)
+        throws TechnicalException;
+
+    List<PortalNavigationItem> findAllByAreaAndEnvironmentId(PortalNavigationItem.Area area, String environmentId)
+        throws TechnicalException;
```

Comment:
I would keep `findAllByAreaAndEnvironmentId` and merge the two others into `findAllByAreaAndEnvironmentIdAndParentId`

Reply:
I know this, that's why in ly other comment I suggested
```
            if (parentId != null) {
                sql += " and parent_id = ?";
            } else {
                sql += " and parent_id is null";
            }
```